### DISCUSSION TITLE
Hierarchical Model Help

### DISCUSSION THREAD

**Discussion by aky9** [lnccbrown/HSSM#260] [Q&A]
Hey! First off, thank you for the new release of HSSM. I can tell a lot of work was put in to its creation! Now for the question. I am relatively new to working with Hierarchical Drift Diffusion Models, but I will lay out what I am trying to accomplish, but I will be a bit vague for privacy reasons. I have two datasets, one from population x and another from population y.  Each dataset contains reaction times, response, block type, and trial type for n number of participants. There are two block types and three trial types. 

This is where the questions come in for fitting this data to a Hierarchical Drift Diffusion Model. To my understanding, I would need to specify the four basic parameters (v,z,a,t) for every combination of population type, block type, and trial type. For this dataset, that would be 3 trial types by 2 block types by 2 populations equaling out to be 12 different versions of each parameter (12 v's, 12 z's etc.). The issue is, I am not seeing an obvious way of specifying this in the HSSM documentation. My assumption is I would define some probability distribution for each parameter as priors and start the model. Then, it would give me the assorted summary stats etc. 

So, is my understanding wrong? If so, what would be the most appropriate way to approach this? If my understanding is correct, how could I do this in HSSM?

Please let me know if I can clarify any details, and thanks for any help!

**Comment by AlexanderFengler** [comment]
Hey, 

sorry for the late reply, we were out on summer holidays. 
You can proceed in multiple ways. 

The simplest, way is to fit the model separately for each of the blocks and trial types, but this is probably not what you want. 

You can also proceed with a targeted hypothesis. E.g. you may consider t, a, z fixed across trial-types and blocks, but assume that v changes.

In this case you can specify a regression for v (you can check the documentation for more details on how to specify the regression precisely) with a (maximal) formula something like this:

v ~ intercept + trial_type + block

Across the two populations, you probably want to specify separate models.
You can then compare the posteriors and posterior predictives across populations using the returned inference data objects.
